file/file.go

```diff
@@ -0,0 +1,70 @@
+// Package file - Handle file operations
+package file
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+)
+
+const filename = "tasks.json"
+
+// File - file generic struct. T generic.
+type File[T any] struct{}
+
+// New - instantiates a File struct.
+func New[T any]() *File[T] {
+	return &File[T]{}
+}
+
+// Write - writes tasks into a JSON file
+func (f *File[T]) Write(data *[]T) error {
+	asJSON, err := json.Marshal(data)
```

Comment:
It's good you learned naming (I mean JSON is all upper case as abbreviation).

But for a variable that is only used in two consecutive lines if a small function it would be enough more general name or even one letter name. Although it's not strict rule.